presto-main/src/test/java/com/facebook/presto/type/TestTimestamp.java

```diff
@@ -237,6 +243,49 @@ public void testCastFromSlice()
     }
 
     @Test
+    public void testCastFromSliceFailsOnTimestampWithTimeZone()
+    {
+        if (connectorSession().isLegacyTimestamp()) {
+            // we expect those not to fail in legacy mode
+            return;
+        }
+
+        functionAssertions.assertFunctionThrowsPrestoException(
+                "cast('2001-1-22 03:04:05.321 +07:09' as timestamp)",
+                INVALID_CAST_ARGUMENT,
+                "Value cannot be cast to timestamp: 2001-1-22 03:04:05.321 +07:09");
```

Comment:
Could this message be more informative? Like "... because it contains tz info" ?